================
Data Categories
================


Due to the extensive collection of data made available by the ISB-CGC the datasets hosted in ISB-CGC Google BigQuery is organized into four categories in our ISB-CGC BIgQuery Table Search user interface, **Processed-omics datasets**, **Genomic Reference Database**, **File Metadata**, and **Clinical Biospecimen Data**. 

A dataset is made up of multiple tables, and can be uniquely identified based on the Google Cloud Platform (GCP) project name (in this case **isb-cgc**), and the dataset name, separated by a colon (or a period, in standard SQL),  *eg* ``isb-cgc.TCGA_bioclin_v0``.  Tables are uniquely identified by appending the table name,
preceded by a period, *eg* ``isb-cgc.TCGA_bioclin_v0.Clinical``.

Processed-omics Datasets
========================

Processed data primarily from the GDC for TCGA, TARGET and CCLE (i.e. raw data that has gone through GDC pipeline processing e.g. gene expression, miRNA expression, copy number, somatic mutations, methylation)

DNA methylation
----------------

Processed data was generated using Illumina Human Methylation 27k and 450k platforms and broken down by chromosome. The entire dataset is all available, which includes all chromosomes in one table.

Somatic Mutations
------------------

need text. 

mRNA Expression
----------------

Gene expression data was generated from the Affymetrix Genome-Wide Human SNP Array 6.0 analysis, ranked Hg38 gene expression data,  Illumina GA, HiSeq sequencing platforms, and the Genotype-Tissue Expression (GTEx) Project.  

miRNA Expression
-----------------

Data was generated by the comparison of individual reads to sequence feature annotations in miRBase v21 and UCSC, Illumina GA, and HiSeq sequencing platforms. 

Protein Quantification
-------------------------

Hasn't been updated since 2017****

Copy Number
------------



Genomic Reference Database
===========================

Genomic information that can be used to cross-reference against processed-omics data tables (e.g. COSMIC, ClinVar, cytoBand, dbSNP)

DNA methylation
----------------

Platform reference annnotations, lift over, and liftover to hg38 coordinates workflow is alailable.  Also, includes an extensive characterization of probes on multiple microarray workflows. 

Somatic Mutations
------------------

Reference for human single nucleotide variations, microsatellites, small-scale insertions, and deletions from multiple tools and databases including Kaviar database. 

Genome Annotation
------------------

Genome Annotation data was loaded from multiple Ensembl gene-set versions and multiple GENCODE reference gene set versions. 

Pathway annotations data has been loaded from multiple ENSEMBL to All pathways mapping workflows from REACTOME, Badar Lab's EnrichmentMap workflow, and WikiPathways.

Related Genome Ontology data is provided by the GO Consortium  and the Gene Ontology Resource. 

This also includes miRNA Expression annotations and target interactions loaded from the miRBase database and the miRTarBase database.

File Metadata
==============

Information about raw data files including Google Cloud Storage paths (e.g. tables with information about files available at the GDC, including GCS paths, creation dates, file uuids, etc.)

Clinical Biospecimen Data
==========================

Patient case and sample information (includes clinical tables with patient demographic data, and biospecimen data with detailed sample information).

Clinical
--------

Patient diagnosis, exposures, and demographic data along with Clinical Annotation can be found. 

Biospecimen
------------

Information on cases tissue sample types, along with information on the generation process of data. 

