******************
About the ISB-CGC
******************


The ISB-CGC platform is built on top of the Google Cloud Platform and has been designed to make the TCGA data as accessible as possible to a wide range of users. For the programmatic users, this includes complete access to the tools that Google is pioneering to allow users to scale-up their analyses on the Google infrastructure using a variety of means.


The ISB-CGC provides interactive and programmatic access to all data hosted by institutes such as the Genomic Data Commons GDC and the Welcome Trust Sanger Institute, leveraging many aspects of the Google Cloud Platform including BigQuery, Compute Engine, App Engine, and Google Cloud Storage.  Open-access clinical and biospecimen information for all TCGA, TARGET, CCLE, COSMIC, and FM patients and samples, combined with the analyzed summary data and genomic reference and platform-annotation sources are stored in BigQuery, enabling fast SQL-like queries against the entire dataset.  Controlled-access DNA and RNA sequence data is available to dbGaP-authorized users in the original BAM and FASTQ file formats for TCGA, TARGET, FM, CTSP, VAREPOP, and NCICCR-DLBCL data at this time.

The ISB-CGC aims to serve the needs of a broad range of cancer researchers 
ranging from scientists or clinicians who prefer to use an interactive 
web-based application to access and explore the rich TCGA, TARGET, and CCLE datasets, to 
computational scientists who want to write their own custom scripts using 
languages such as R or Python, accessing the data through APIs, to algorithm 
developers who want to spin up thousands of virtual machines to rapidly 
analyze hundreds of terabytes of sequence data.  The ISB-CGC allows scientists 
to interactively define and compare cohorts, examine the underlying molecular 
data for specific genes or pathways of interest, and share insights with 
collaborators around the globe.  

